packages/api/src/router/activity.ts

```diff
@@ -0,0 +1,78 @@
+import { z } from "zod";
+
+import { adminProcedure, createTRPCRouter, protectedProcedure } from "../trpc";
+
+export const activityRouter = createTRPCRouter({
+  getSchedule: protectedProcedure
+    .input(z.object({ day: z.date() }))
+    .query(async ({ ctx, input }) => {
+      return await ctx.db.activity.findMany({
+        where: {
+          day: {
+            gte: new Date(input.day),
+            lte: new Date(input.day.setDate(input.day.getDate() + 7)),
```

Comment:
```suggestion
            gte: input.day,
            lte: new Date(input.day.getDate() + 7),
```

Reply:
this passed typechecks for me -- is there a reason why you said that it didn't work? (this is very last thing for PR then will approve 😰 )

Reply:
Yea! It's really weird it doesn't have a typecheck error, but with testing it becomes incorrect. 
![Screenshot 2023-10-29 at 2 35 58 PM](https://github.com/uoftblueprint/centre-for-dreams/assets/113125436/a43ae6e3-79dc-4316-8714-71966f8a604b)


With `input.day`:
<img width="205" alt="Screenshot 2023-10-29 at 2 29 26 PM" src="https://github.com/uoftblueprint/centre-for-dreams/assets/113125436/b644c609-66d0-48ab-9890-0a478fcc6508">

With `new Date(input.day)`:
<img width="212" alt="Screenshot 2023-10-29 at 2 31 08 PM" src="https://github.com/uoftblueprint/centre-for-dreams/assets/113125436/645f17a9-8758-4cd8-89a9-cee76ef91ff3">

Table for reference: 
<img width="611" alt="Screenshot 2023-10-29 at 2 31 29 PM" src="https://github.com/uoftblueprint/centre-for-dreams/assets/113125436/5d402765-59a4-4dec-9457-8592a14dd2bc">

Reply:
Also, `new Date(input.day.getDate() + 7)` doesn't work either! Get date only returns the day of the month, so it wouldn't update the month or year. It's ugly but we'd either do it with set date or `new Date(input.day.getTime() + 7 * 24 * 60 * 60 * 1000)`

Reply:
We could import `date-fns` or some other library that makes it easier to manipulate dates but idk.

Reply:
thanks for the good testing -- I think best option is to use date-fns here, since it will probably make the code more readable. You will just need to `pnpm add date-fns` in the `packages/api` folder to add it as a dependency

Reply:
Omg it's so much ✨cleaner✨! It also fixed the weird issue with input.day for whatever reason

Reply:
I think that `input.day` wasn't working originally because the value was being modified in the `setDate` call. `date-fns` creates a new date object for each of its functions, which is why using `input.day` works now

Reply:
Ohhhhh yea that makes sense